substrate/frame/assets/src/lib.rs

```diff
@@ -404,6 +407,9 @@ pub mod pallet {
 		/// used to set up auto-incrementing asset IDs for this collection.
 		type CallbackHandle: AssetsCallback<Self::AssetId, Self::AccountId>;
 
+		/// A type used to verify merkle proofs used for distributions.
+		type VerifyExistenceProof: VerifyExistenceProof<Hash: Parameter + MaxEncodedLen + Default>;
```

Comment:
```suggestion
		type VerifyExistenceProof: VerifyExistenceProof<Hash = <Self as frame_system>::Hash>;
```

or sth along these lines?